pkg/storage/interfaces.go

```diff
@@ -62,34 +64,13 @@ type MatchValue struct {
 // to that function.
 type TriggerPublisherFunc func(obj runtime.Object) []MatchValue
 
-// Filter is interface that is used to pass filtering mechanism.
-type Filter interface {
-	// Filter is a predicate which takes an API object and returns true
-	// if and only if the object should remain in the set.
-	Filter(obj runtime.Object) bool
-	// For any triggers known to the Filter, if Filter() can return only
-	// (a subset of) objects for which indexing function returns <value>,
-	// (<index name>, <value> pair would be returned.
-	//
-	// This is optimization to avoid computing Filter() function (which are
-	// usually relatively expensive) in case we are sure they will return
-	// false anyway.
-	Trigger() []MatchValue
-}
-
-// Everything is a Filter which accepts all objects.
-var Everything Filter = everything{}
-
-// everything is implementation of Everything.
-type everything struct {
-}
-
-func (e everything) Filter(runtime.Object) bool {
-	return true
-}
+// FilterFunc takes an API object and returns true if the object satisfies some requirements.
+type FilterFunc func(obj runtime.Object) bool
```

Comment:
Why are you keeping this type at all instead of making everything use SelectionPredicates?